perceval/backend.py

```diff
@@ -161,6 +214,7 @@ def metadata(self, item):
             'origin': self.origin,
             'uuid': uuid(self.origin, self.metadata_id(item)),
             'updated_on': self.metadata_updated_on(item),
+            'classified_fields_filtered': self.classified_fields if filter_classified else None,
```

Comment:
I'm not sure if this should be included to the item here or later only if `filter_classified` is true. Logically, it makes sense to add it where it is now, but in practice it implies that we should re-create all raw indexes we have (to avoid mixing old items created before this change and new ones with the `classified_fields_filtered` attribute)

If we add the field only if `filter_classified` is true, this would only affect the raw indexes for the customer(s) that asked for this feature.

Reply:
I thought of that but after a long discussion with myself, I still think all items generated with a given version should have the same fields, at least in terms of those fields we control. I know it is verbose for the cases where is not in use, but it allows to see nothing was filtered, instead of wondering why this field is available on these items and not in others generated with the same Perceval version.

In terms of storing these items I would also like to have all items with the same data but sometimes it isn't possible.
On the other hand, GrimoireLab uses a schemaless database (Elastic Search) to store items. This change won't affect already created indexes. New items will be added to existing indexes without problems.

Moreover, take into account, this field only works as reference/debugging. To know if data was filtered from the original or not. It shouldn't be used in queries, etc.

Reply:
Thank you for the clarification, it sounds reasonable.

> On the other hand, GrimoireLab uses a schemaless database (Elastic Search) to store items. This change won't affect already created indexes. New items will be added to existing indexes without problems.

Yes, this change won't affect existing indexes (for the schemaless structure), but we will have items with different meta information. I guess we can live with this discrepancy for a while.